fastNLP/core/trainer.py

```diff
@@ -139,6 +141,26 @@ def _train_step(self, data_iterator, network, **kwargs):
                 logger.info(print_output)
             step += 1
 
+    def cross_validate(self, network, train_data_cv, dev_data_cv):
+        """Training with cross validation.
+
+        :param network: the model
+        :param train_data_cv: four-level list, of shape [num_folds, num_examples, 2, ?]
+        :param dev_data_cv: four-level list, of shape [num_folds, num_examples, 2, ?]
+
+        """
+        if len(train_data_cv) != len(dev_data_cv):
+            logger.error("the number of folds in train and dev data unequals {}!={}".format(len(train_data_cv),
+                                                                                            len(dev_data_cv)))
+            raise RuntimeError("the number of folds in train and dev data unequals")
+        n_fold = len(train_data_cv)
+        logger.info("perform {} folds cross validation.".format(n_fold))
+        for i in range(n_fold):
+            print("CV:", i)
+            logger.info("running the {} of {} folds cross validation".format(i + 1, n_fold))
+            network_copy = copy.deepcopy(network)
+            self.train(network_copy, train_data_cv[i], dev_data_cv[i])
```

Comment:
I think self.validate should set to be true in this function if you want to use validation in train()

Reply:
yes, will fix